src/com/facebook/buck/swift/SwiftCompileStep.java

```diff
@@ -43,16 +44,22 @@
   private final ImmutableMap<String, String> compilerEnvironment;
   private final ImmutableList<String> compilerCommand;
   private final Optional<AbsPath> argsFile;
+  private final Optional<AbsPath> vfsOverlayFile;
+  private final Optional<VFSOverlay> vfsOverlay;
 
   SwiftCompileStep(
       AbsPath compilerCwd,
       Optional<AbsPath> argsFile,
       Map<String, String> compilerEnvironment,
-      Iterable<String> compilerCommand) {
+      Iterable<String> compilerCommand,
+      Optional<VFSOverlay> vfsOverlay,
+      Optional<AbsPath> vfsOverlayFile) {
```

Comment:
I think we can remove the `vfsOverlay` parameter here and simply rely on `vfsOverlayFile`.

Reply:
This step is to write `vfsOverlay` to `vfsOverlayFile`, so we need to have both. Before this method, `vfsOverlayFile` is just a path and no content. We can only write to an output file in a subclass of `Step`.